sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -69,6 +69,38 @@ def merge(
 
 
 class LogicalReplaceQueryMixin(EngineAdapter):
+    @classmethod
+    def overwrite_target_from_temp(
+        cls,
+        engine_adapter: EngineAdapter,
+        query: Query,
+        columns_to_types: t.Dict[str, exp.DataType],
+        target_table: TableName,
+    ) -> None:
+        """
+        Overwrites the target table from the temp table. This is used when the target table is self-referencing.
+        """
+        with engine_adapter.temp_table(
+            exp.select(*columns_to_types).from_(target_table),
+            target_table,
+            columns_to_types,
+        ) as temp_table:
+
+            def replace_table(
+                node: exp.Expression, curr_table: exp.Table, new_table: exp.Table
+            ) -> exp.Expression:
+                if isinstance(node, exp.Table) and quote_identifiers(node) == quote_identifiers(
```

Comment:
@treysp FYI I think this was another bug outside of the spark issue. `node == curr_table` could say that two tables don't match if one doesn't have quotes and the other does. So we quote them both now when comparing.

Reply:
Good catch - thanks